test/local/account_routes.js

```diff
@@ -1129,13 +1208,163 @@ test('/account/login', function (t) {
+          t.equal(response.verificationReason, 'signup', 'verificationReason is signup')
+          t.equal(response.emailSent, true, 'email sent')
+        }).then(function () {
+          mockRequest.query.sendEmailIfUnverified = undefined
+          mockMailer.sendVerifyCode.reset()
+        })
+      })
+
+      t.test('with `sendEmailIfUnverified` param, false', function (t) {
+        mockRequest.query.sendEmailIfUnverified = false
+        return runTest(route, mockRequest, function (response) {
+          t.equal(mockMailer.sendVerifyCode.callCount, 0, 'mailer.sendVerifyCode was not called')
+          t.equal(mockMailer.sendVerifyLoginEmail.callCount, 0, 'mailer.sendVerifyLoginEmail was not called')
+          t.equal(mockMailer.sendNewDeviceLoginNotification.callCount, 0, 'mailer.sendNewDeviceLoginNotification was not called')
+          t.equal(response.verified, false, 'response indicates account is unverified')
+          t.equal(response.verificationMethod, 'email', 'verificationMethod is email')
+          t.equal(response.verificationReason, 'signup', 'verificationReason is signup')
+          t.equal(response.emailSent, false, 'email sent')
+        }).then(function () {
+          mockRequest.query.sendEmailIfUnverified = undefined
+        })
+      })
+
+      t.test('not from content server', function (t) {
+        mockRequest.query.sendEmailIfUnverified = undefined
+        mockRequest.payload.metricsContext = undefined
+        return runTest(route, mockRequest, function (response) {
+          t.equal(mockMailer.sendVerifyCode.callCount, 1, 'mailer.sendVerifyCode was called')
+          t.equal(mockMailer.sendVerifyLoginEmail.callCount, 0, 'mailer.sendVerifyLoginEmail was not called')
+          t.equal(mockMailer.sendNewDeviceLoginNotification.callCount, 0, 'mailer.sendNewDeviceLoginNotification was not called')
+          t.equal(response.verified, false, 'response indicates account is unverified')
+          t.equal(response.verificationMethod, 'email', 'verificationMethod is email')
+          t.equal(response.verificationReason, 'signup', 'verificationReason is signup')
+          t.equal(response.emailSent, true, 'email not sent')
+        }).then(function () {
+          mockMailer.sendVerifyCode.reset()
+        })
+      })
+    })
   })
 
   t.test('creating too many sessions causes an error to be logged', function (t) {
+    mockDB.emailRecord = function () {
+      return P.resolve({
+        authSalt: crypto.randomBytes(32),
+        data: crypto.randomBytes(32),
+        email: 'test@mozilla.com',
+        emailVerified: true,
+        kA: crypto.randomBytes(32),
+        lastAuthAt: function () {
+          return Date.now()
+        },
+        uid: uid,
+        wrapWrapKb: crypto.randomBytes(32)
+      })
+    }
     mockDB.sessions = function () {
       return P.resolve(new Array(200))
     }
     mockLog.error = sinon.spy()
+    mockRequest.app.clientAddress = '63.245.221.32'
```

Comment:
Added this so that geodb does not log invalid ip error. I suspect a previous test case is modifying the ip address in mockRequest. Could possibly be fixed by https://github.com/mozilla/fxa-auth-server/issues/1448.